src/gluonnlp/model/transformer.py

```diff
@@ -476,16 +479,30 @@ def hybrid_forward(self, F, inputs, states=None, valid_length=None, position_wei
             mask = states[-2]
         else:
             mask = None
+
+        all_encodings_outputs = []
         additional_outputs = []
         for cell in self.transformer_cells:
             outputs, attention_weights = cell(inputs, mask)
             inputs = outputs
+            if self._output_all_encodings:
+                outputs_masked = outputs
+                if valid_length is not None:
+                    outputs_masked = F.SequenceMask(outputs, sequence_length=valid_length,
+                                                    use_sequence_length=True, axis=1)
+                all_encodings_outputs.append(outputs_masked)
```

Comment:
```python
            if self._output_all_encodings:
                if valid_length is not None:
                    outputs= F.SequenceMask(outputs, sequence_length=valid_length,
                                                    use_sequence_length=True, axis=1)
                all_encodings_outputs.append(outputs)
```
It is better if the names are consistent. Maybe it's more concise.

Reply:
Fixed